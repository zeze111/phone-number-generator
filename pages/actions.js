import fetch from "isomorphic-unfetch";

export const generateNumbers = limit =>
  fetch("http://localhost:3000/telecomms/", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json"
    },
    body: JSON.stringify({ limit })
  }).then(response => response.json());

export const listNumbers = (offset, limit) =>
  fetch(
    `http://localhost:3000/telecomms/phone-numbers?limit=${limit}&offset=${offset}`
  ).then(response => response.json());

export const sortNumbers = (order, offset, limit)  =>
  fetch(
    `http://localhost:3000/telecomms/phone-numbers/sort/${order}?limit=${limit}&offset=${offset}`
  ).then(response => response.json());

export const clearNumbersStorage = () =>
  fetch("http://localhost:3000/telecomms/phone-numbers", {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json"
    }
  }).then(response => response.json());
