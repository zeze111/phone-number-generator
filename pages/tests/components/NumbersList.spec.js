import React from "react";
import { shallow } from "enzyme";
import NumbersList from "../../components/NumbersList";
import { listNumbers, sortNumbers, clearNumbersStorage } from "../../actions";

jest.mock("../../actions", () => ({
  listNumbers: jest.fn(() => ({
    numbers: ["012", "123", "234", "345"],
    total: 4
  })),
  sortNumbers: jest.fn(() => ({
    numbers: ["012", "123", "234", "345"],
    response: {}
  })),
  clearNumbersStorage: jest.fn(() => ({
    message: "All numbers have been cleared"
  }))
}));

describe("NumbersList", () => {
  it("renders the component", () => {
    const wrapper = shallow(<NumbersList />);
    wrapper.setState({ numbers: ["012", "123", "234", "345"] });
    expect(wrapper).toMatchSnapshot();
  });

  describe("orderList", () => {
    const wrapper = shallow(<NumbersList />);
    const button = wrapper.find(".order-button");

    button.simulate("click");
    expect(sortNumbers).toHaveBeenCalledWith("desc", 0, 3);
  });

  describe("clearList", () => {
    const wrapper = shallow(<NumbersList />);
    const button = wrapper.find(".clear-button");

    button.simulate("click");
    expect(clearNumbersStorage).toHaveBeenCalled();
  });
});
