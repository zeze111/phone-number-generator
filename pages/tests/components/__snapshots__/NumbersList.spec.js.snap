// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersList renders the component 1`] = `
<div>
  <div
    className="generate-content"
  >
    <h3>
       List of Numbers Generated 
    </h3>
    <div>
      <button
        className="order-button"
        onClick={[Function]}
        type="button"
      >
        Order
      </button>
      asc
      <button
        className="clear-button"
        onClick={[Function]}
        type="button"
      >
        Clear
      </button>
    </div>
    <div
      className="list-numbers"
    >
      <ul>
        <li
          className="numbers"
          key="0"
        >
          012
        </li>
        <li
          className="numbers"
          key="1"
        >
          123
        </li>
        <li
          className="numbers"
          key="2"
        >
          234
        </li>
        <li
          className="numbers"
          key="3"
        >
          345
        </li>
      </ul>
    </div>
    <div>
      <p>
         Total Numbers Generated: 
      </p>
      <p>
         Minimum Number Generated: 
      </p>
      <p>
         Maximum Number Generated: 
      </p>
    </div>
  </div>
</div>
`;
